code/z_adventurezones/safehouse.dm

```diff
@@ -0,0 +1,708 @@
+		name = "Safehouse (asteroid doors)"
+		icon_state = "green"
+		force_fullbright = 1
+
+//TURFS
+
+/turf/unsimulated/floor/circuit/green/green_CO2
+	icon_state = "circuit-green"
+	RL_LumR = 0
+	RL_LumG = 0.3
+	RL_LumB = 0
+	carbon_dioxide = 90
+	nitrogen = 7
+	oxygen = 3
+
+/turf/unsimulated/floor/circuit/vintage/vintage_CO2
+	icon_state = "circuit-vint1"
+	RL_LumR = 0.1
+	RL_LumG = 0.1
+	RL_LumB = 0.1
+	carbon_dioxide = 90
+	nitrogen = 7
+	oxygen = 3
+
+//ASTEROID DOOR PUZZLE
+
+/obj/machinery/door/poddoor/blast/asteroid
+	name = "asteroid"
+	id = "podbay_saferoom"
+	icon = 'icons/obj/adventurezones/safehouse.dmi'
+	icon_state = "adoorsingle1"
+	desc = "A free-floating mineral deposit from space."
+	icon_base = "adoor"
+	doordir = "single"
+
+	flags = FPRINT | IS_PERSPECTIVE_FLUID | ALWAYS_SOLID_FLUID //The poddoors aren't inherently fullbright, need a suitable turf or area underneath.
+
+	podbay_autoclose
+		autoclose = 1
+
+		asteroid_horizontal
+			name = "podbay (asteroid)"
+			id = "podbay_saferoom"
+			dir = NORTH
+
+			vertical
+				dir = EAST
+
+/obj/machinery/door_control/podbay/suspiciousdebris/
+	name = "suspicious debris"
+	id = "podbay_safehouse"
+	icon = 'icons/obj/adventurezones/safehouse.dmi'
+	icon_state = "doorctrl0"
+	desc = "Someone appears to have bolted this useless heap of junk to an asteroid."
+	layer = OBJ_LAYER
+	cooldown = 6 DECI SECONDS //To try and sync animation without redefining attack_hand.
+
+//SAFE ROOM PUZZLE
+
+/mob/living/carbon/human/dead_exec
+	name = "Jean Rockefeller" //This guy shows up as invisible in StrongDMM, not a clue why.
+	real_name = "Jean Rockefeller"
+	gender = MALE
+	interesting = "This guy seems like he's been through a lot."
+
+	initializeBioholder() //We need bioholder data intialised so we can use it elsewhere.
+		bioHolder.ownerName = name
+		bioHolder.mobAppearance.customization_first = new /datum/customization_style/moustache/vandyke
+		bioHolder.mobAppearance.customization_first_color = "#241200"
+		bioHolder.mobAppearance.customization_second = new /datum/customization_style/none
+		bioHolder.mobAppearance.customization_second_color = "#241200"
+		bioHolder.mobAppearance.customization_third = new /datum/customization_style/none
+		bioHolder.mobAppearance.customization_third_color = "#241200"
+		bioHolder.mobAppearance.e_color = "#363978"
+		bioHolder.mobAppearance.s_tone = "#FFCC99"
+		bioHolder.age = 52
+		bioHolder.bloodType = "O+"
+		bioHolder.mobAppearance.gender = "male"
+		bioHolder.mobAppearance.underwear = "none"
+		bioHolder.mobAppearance.u_color = "#FFFFFF"
+		bioHolder.Uid = bioHolder.CreateUid()
+		bioHolder.uid_hash = md5(bioHolder.Uid)
+		. = ..()
+
+obj/item/reagent_containers/iv_drip/dead_exec
+	desc = "A bag filled with someone's blood. It's labelled 'Jean Rockefeller'."
+	icon_state = "IV-blood"
+	mode = 1
+	initial_reagents = "blood"
+
+	New()
+		..()
+		START_TRACKING
+
+	disposing()
+		STOP_TRACKING
+		..()
+
+/obj/machinery/handscanner/bio_handscanner
```

Comment:
Is there any reason for this to be a child of handscanner if so much of the behavior is overridden?

Reply:
Not really, was just how code was built up as I learned. It’s no longer a child of hand scanner